cmd/inspect/block.go

```diff
@@ -222,27 +222,80 @@ var blockCmd = &cobra.Command{
 	},
 }
 
-func prettyPrintCBOR(indent string, data []byte) {
+func prettyPrintCBOR(indent string, kind string, data []byte) {
 	var body interface{}
 	if err := cbor.Unmarshal(data, &body); err != nil {
-		fmt.Printf("%s%s\n", indent, base64.StdEncoding.EncodeToString(data))
+		rawPrintData(indent, kind, data)
 		return
 	}
 
-	prettyPrintStruct(indent, data, body)
+	prettyPrintStruct(indent, kind, data, body)
 }
 
-func prettyPrintStruct(indent string, data []byte, body interface{}) {
+func encode(in interface{}) (interface{}, error) {
+	v := reflect.ValueOf(in)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	switch v.Kind() {
+	case reflect.Slice:
+		// Walk slices.
+		for i := 0; i < v.Len(); i++ {
+			elem := v.Index(i)
+			encoded, err := encode(elem.Interface())
+			if err != nil {
+				return nil, err
+			}
+			elem.Set(reflect.ValueOf(encoded))
+		}
+		return v.Interface(), nil
```

Comment:
Is there an easy way to output the binary value encoded as hex instead of Base64?

Reply:
Done.